### PR TITLE
Add machine type category

### PR DESCRIPTION
**Pull request by tzvatot**
@nimrodshn @igoihman @zgalor  PTAL

### CODE REVIEW

File: model/clusters_mgmt/v1/machine_type_type.model [hunk 1]
```diff
@@ -19,6 +19,10 @@ class MachineType {
 	// Human friendly identifier of the machine type, for example `r5.xlarge - Memory Optimized`.
 	Name String
 
+	// The category which the machine type is suitable for.
+	// Options: `general_purpose`, `memory_optimized`, `compute_optimized`.
+	Category String
```
Comment (zgalor) [inline model/clusters_mgmt/v1/machine_type_type.model:24]: If it only accepts 3 values, why is it a string and not a type, like we do for state?

Comment (tzvatot) [reply model/clusters_mgmt/v1/machine_type_type.model]: Make sense. PreTAL.

File: model/clusters_mgmt/v1/machine_type_category_type.model [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Machine type category
+enum MachineTypeCategory {
+	// General Purpose machine type
```
Comment (zgalor) [inline model/clusters_mgmt/v1/machine_type_category_type.model:19]: The parser needs the comments to end with a `.` in order to properly generate the openapi.yaml file

File: model/clusters_mgmt/v1/machine_type_category_type.model [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	GeneralPurpose
+
+	// Memory Optimized machine type
```
Comment (zgalor) [inline model/clusters_mgmt/v1/machine_type_category_type.model:22]: same

File: model/clusters_mgmt/v1/machine_type_category_type.model [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	MemoryOptimized
+
+	// Compute Optimized machine type
```
Comment (zgalor) [inline model/clusters_mgmt/v1/machine_type_category_type.model:25]: same

Comment (tzvatot) [reply model/clusters_mgmt/v1/machine_type_category_type.model]: Fixed

File: model/clusters_mgmt/v1/machine_type_category_type.model (hunk 3 above)
Comment (tzvatot) [reply model/clusters_mgmt/v1/machine_type_category_type.model]: Fixed

File: model/clusters_mgmt/v1/machine_type_category_type.model (hunk 2 above)
Comment (tzvatot) [reply model/clusters_mgmt/v1/machine_type_category_type.model]: Fixed. Why is this not covered in the formatter or the UT?

**Comment by tzvatot** [comment]
@zgalor PreTAL

File: model/clusters_mgmt/v1/machine_type_category_type.model (hunk 2 above)
Comment (zgalor) [reply model/clusters_mgmt/v1/machine_type_category_type.model]: Don't know. Can you add it?

**Comment by zgalor** [comment]
LGTM

**Pull request merged**